spec/controllers/api/v1/project_preferences_controller_spec.rb

```diff
@@ -137,4 +137,34 @@
 
     it_behaves_like "is creatable"
   end
+
+  describe "#update_settings", :focus do
+    let!(:project) { create(:project, owner: authorized_user) }
+    let!(:upp) { create(:user_project_preference, project: project) }
+    let(:settings_params) do
+      {
+        user_id: upp.user_id,
+        project_id: project.id,
+        settings: { workflow_id: 1234 }
+      }
+    end
+
+    let(:scopes) { %w(project_preference) }
```

Comment:
Remove the `let(:scopes) { %w(project_preference) }` def here and you'll be allowed into the controller before_action filter. I'm pretty sure that the scopes here is 422'ing and not entering the controller (due to scopes restriction via doorkeeper). Removing this and still 422 is due to the internal controller, most likely the json schema (check the response body).